app/models/concerns/foreman_kernel_care/host_managed_extensions.rb

```diff
@@ -69,8 +98,36 @@ def sync_kernel_associations(new_patched_kernel_ids, delete_ids)
       end
     end
 
+    def find_deb_kernel_packages(old_version)
+      installed_debs.select do |p|
+        p.name.include?('generic') && p.version.include?(old_version)
+      end
+    end
+
+    def create_new_deb_kernel_packages(packages, new_version, old_version)
+      delete_ids = packages.map(&:id)
+      new_kernels = packages.map do |p|
+        name = if p.name.include?(old_version)
+                 p.name.sub(old_version, new_version)
+               else
+                 p.name
+               end
+        ::Katello::InstalledDeb.find_or_create_by(name: name, architecture: p.architecture, version: new_version)
+      end
+      sync_deb_kernel_associations(new_kernels.map(&:id), delete_ids)
```

Comment:
Wouldn't it be good to check if new_kernels and delete_ids is set before calling the method? Maybe they are empty? I did not verify if it's ok if they are empty.

Reply:
?